src/backend/chat_service/chat_service.py

```diff
@@ -94,6 +100,74 @@ async def add_customer(request: Request):
         "customer_guid": customer_guid
     }
 
+
+#API endpoint for UploadFile api
+@app.post("/uploadFile",tags=["File Management"])
+async def upload_File(request: Request, customer_guid: str = Form(...), file:UploadFile=File(...)):
+    logger.debug(f"Entering upload_file() with Correlation ID:{request.state.correlation_id}")
+    try:
+        logger.info(f"uploading file '{file.filename} of type '{file.content_type}'")
+
+        #call MinioManager to Upload the file
+        minio_manager.upload_file(
+            bucket_name=customer_guid,
+            filename=file.filename,
+            file_data=file.file
+        )
+        logger.info(f"File '{file.filename}' uploaded to bucket '{customer_guid}' successfully.")
+        return {"message":"File uploaded SuccessFully"}
+    except Exception as e:
+        logger.error(f"Error in file upload:{e}")
+        raise HTTPException(status_code=500,detail="Error uploading the file")
+    finally:
+        logger.debug(f"Exiting upload_file() with Correlation ID:{request.state.correlation_id}")
+
+
+@app.get("/listfiles", tags=["File Management"])
+async def list_files(request: Request, customer_guid: str):
+    logger.debug(f"Entering list_files() with Correlation ID: {request.state.correlation_id}")
+    try:
+        # Call MinioManager to get the file list
+        file_list = minio_manager.list_files(bucket_name=customer_guid)
+        if file_list:
+            logger.info(f"Files retrieved for bucket '{customer_guid}':{file_list}")
+        else:
+            logger.info(f"No files found in bucket '{customer_guid}'")
+
+        return {"files": file_list}
+    except Exception as e:
+        logger.error(f"Error listing files:'{customer_guid}': {e}")
+        raise HTTPException(status_code=500, detail="Error listing files")
+    finally:
+        logger.debug(f"Exiting list_files() with Correlation ID: {request.state.correlation_id}")
+
+
+@app.get("/downloadfile", tags=["File Management"])
+async def download_file(request:Request, customer_guid:str, filename:str):
+    logger.debug(f"Entering download_file() with Correlation ID:{request.state.correlation_id}")
+    try:
+        file_data=minio_manager.download_file(
```

Comment:
(Option 1) This program might crash for larger files, can we obtain chunks from minio and then stream it back so that the file chunks are smaller. if we get the entire file data in one function call, our service will grow large in size and crash

**OR**

(Option 2) can we get a file_object returned from minio which can be used to stream the file bytes to the client

try for option 2 

If its not feasible, move to option 1